src/python/zquantum/core/circuit/_gate_test.py

```diff
@@ -1,7 +1,184 @@
 import unittest
 import os
 from ._gate import Gate
+from ._gateset import COMMON_GATES, UNIQUE_GATES
+from ._qubit import Qubit
+from sympy import Symbol
+import qiskit
+from math import pi
+
 
 class TestGate(unittest.TestCase):
-    def test_test(self):
-        pass
\ No newline at end of file
+    def setUp(self):
+        self.two_qubit_gates = ["CNOT", "CZ", "CPHASE", "SWAP"]
+        self.one_parameter_gates = ["PHASE", "Rx", "Ry", "Rz", "CPHASE"]
+
+    def create_gate(self, gate_name, params=None):
+        if gate_name in self.two_qubit_gates:
+            qubit_list = [Qubit(0), Qubit(1)]
+        else:
+            qubit_list = [Qubit(0)]
+        if params is None:
+            params = []
+            if gate_name in self.one_parameter_gates:
+                params = [1.0]
+        gate = Gate(gate_name, qubits=qubit_list, params=params)
+        return gate, qubit_list
+
+    def create_gate_with_symbolic_params(self, gate_name):
+        if gate_name in self.two_qubit_gates:
+            qubit_list = [Qubit(0), Qubit(1)]
+        else:
+            qubit_list = [Qubit(0)]
+        params = []
+        if gate_name in self.one_parameter_gates:
+            params = [Symbol("theta_0")]
+        gate = Gate(gate_name, qubits=qubit_list, params=params)
+        return gate, qubit_list, params
+
+    def test_create_evaluated_gate(self):
+        # Given
+        gate, _ = self.create_gate("Rx")
+        symbolic_gate, _, params = self.create_gate_with_symbolic_params("Rx")
+        symbols_map = [(params[0], 1.0)]
+
+        # When
+        evaluated_symbolic_gate = symbolic_gate.create_evaluated_gate(symbols_map)
+        evaluated_regular_gate = gate.create_evaluated_gate(symbols_map)
+
+        # Then
+        self.assertEqual(gate, evaluated_symbolic_gate)
+        # Check if the params of the initial gate has not been overwritten
+        self.assertEqual(symbolic_gate.params[0], symbols_map[0][0])
+        # Check if evaluating a regular gate returns the exact same gate
+        self.assertEqual(evaluated_regular_gate, gate)
+
+        # Given
+        symbols_map = [("x", 1.0)]
+
+        # When
+        evaluated_symbolic_gate = symbolic_gate.create_evaluated_gate(symbols_map)
+
+        # Then
+        self.assertEqual(evaluated_symbolic_gate, symbolic_gate)
```

Comment:
I feel like there are two different tests here. Maybe split this function up into the "normal" `gate` and `symbolic_gate`

Reply:
Also maybe loop over gates in `COMMON_GATES`?

Reply:
Done in `dd8660d6b835c6f15c0e4da35d39e8da14ffbf78`